src/Microsoft.ML.Data/Transforms/TransformsExtensionsCatalog.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Transforms;
+
+namespace Microsoft.ML
+{
+    using HashDefaults = Transforms.HashEstimator.Defaults;
+
+    /// <summary>
+    /// Extensions for Column Copying Estiamtor.
+    /// </summary>
+    public static class ColumnCopyingCatalog
+    {
+        /// <summary>
+        /// Copies the input column to another column named as specified in <paramref name="outputColumn"/>.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="inputColumn">The source column.</param>
+        /// <param name="outputColumn">The new column, resulting from copying.</param>
+        public static CopyColumnsEstimator CopyColumns(this TransformsCatalog catalog, string inputColumn, string outputColumn)
+            => new CopyColumnsEstimator(CatalogUtils.GetEnvironment(catalog), inputColumn, outputColumn);
+
+        /// <summary>
+        /// Copies the input column, name specified in the first item of the tuple,
+        /// to another column, named as specified in the second item of the tuple.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog</param>
+        /// <param name="columns">The pairs of input and output columns.</param>
+        public static CopyColumnsEstimator CopyColumns(this TransformsCatalog catalog, params (string source, string name)[] columns)
+            => new CopyColumnsEstimator(CatalogUtils.GetEnvironment(catalog), columns);
+
+    }
+
+    /// <summary>
+    /// Extension ColumnConcatenatingEstimator
+    /// </summary>
+    public static class ColumnConcatenatingEstimator
+    {
+        /// <summary>
+        /// Concatenates two columns together.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="inputColumns">The names of the columns to concatenate together.</param>
+        /// <param name="outputColumn">The name of the output column.</param>
```

Comment:
should we sort it?

Reply:
I am not sure I understand.

---
In reply to: [229404762](https://github.com/dotnet/machinelearning/pull/1448#discussion_r229404762) [](ancestors = 229404762)

Reply:
order in function:
 this TransformsCatalog catalog, string outputColumn, params string[] inputColumns

order in summary params is 
catalog
inputColumns,
outputColumn.

---
In reply to: [229410418](https://github.com/dotnet/machinelearning/pull/1448#discussion_r229410418) [](ancestors = 229410418,229404762)

Reply:
ic, thanks!

---
In reply to: [229415251](https://github.com/dotnet/machinelearning/pull/1448#discussion_r229415251) [](ancestors = 229415251,229410418,229404762)